torch/_refs/__init__.py

```diff
@@ -4020,10 +4020,10 @@ def index_select(x: TensorLike, dim: int, index: TensorLike):
     )
     if index.ndim == 0:
         index = index.unsqueeze(0)
-    if x.ndim == 0:
+    if x.numel() == 1:
         # Treat scalars as elements of \R^1
         # We cannot use x[idx] here as it accesses item() (??), hence this awkward construction
```

Comment:
Do we know why this happens?!

Reply:
See https://github.com/pytorch/pytorch/issues/105641

Reply:
Can we add an operator that is like `x[idx]` but instead produces a copy rather than view?